compiler/quilt/test/test_cli.py

```diff
@@ -456,37 +467,70 @@ def setUp(self):
         self.env = os.environ.copy()
         self.env['PYTHON_PATH'] = PACKAGE_DIR
 
-        self.quilt_command = [sys.executable, '-c', 'from quilt.tools import main; main.main()']
+        self.quilt_command = [sys.executable, '-c', 'from quilt.tools import main; main.main()',
+                              'quilt testing']
 
     def tearDown(self):
         # restore the real 'command' module back to the 'main' module
         self._main.command = self.mock_command._target
 
     def execute(self, cli_args):
-        result = {}
+        """Execute a command using the method specified by the environment
 
+        When "QUILT_TEST_CLI_SUBPROC" is set to "True", use a subprocess.
+        Otherwise, call main() directly.
+
+        :returns: dict of return codes and calls made to `command` functions
+        """
         # CLI mode -- actually executes "quilt <cli args>"
         # This mode is preferable, once quilt load times improve.
         if self.env.get('QUILT_TEST_CLI_SUBPROC', '').lower() == 'true':
-            quilt = self.quilt_command
-            env = self.env
-            cmd = quilt + cli_args
-            try:
-                result = json.loads(check_output(cmd, env=env).decode())
-                result['return code'] = 0
-            except CalledProcessError as error:
-                result['return code'] = error.returncode
-            return result
+            return self.execute_cli(cli_args)
         # Fast mode -- calls main.main(cli_args) instead of actually executing quilt
         else:
-            try:
-                self._main.main(cli_args)
-            except SystemExit as error:
-                result['return code'] = error.args[0] if error.args else 0
-            else:
-                result['return code'] = 0
-            result.update(self.mock_command._result)
-            return result
+            return self.execute_fast(cli_args)
+
+    def execute_cli(self, cli_args):
```

Comment:
dumb q: I thought [something like this] was already committed as part of the CLI testing?  sorry if dumb q...

Reply:
Not a dumb question -- that's just the original function, split into two functions, so that I could call the execute_cli() one separately.  I ended up needing to use Popen, anyways, so it was moot.  ..but I left the change in in case some future test has reason to prefer either `execute_cli` or `execute_fast` over using `execute`, which auto-selects cli or fast mode based on environment.